### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by vdboor**
dev-docs topics:

* Improved intro
* Intro to REST framework
* Steaming rendering
* Temporal datasets

End user docs:

* Temporal datasets.
* Split long REST/WFS page into separate pages.
* Fix some generates dataset content.
* Explain ``?_fields=...`` parameter.
* Fix WFS example links.
* Fix date-time field types.

### CODE REVIEW

File: dev-docs/source/dynamic_api.rst [hunk 1]
```diff
@@ -47,10 +56,8 @@ When all viewsets are constructed, reading ``router.urls`` returns all available
...
-.. tip::
-    To debug datasets and use their models, you can reuse the router logic
-    which already created those models. The following can be used inside ``./manage.py shell``::
+Serializer Class Layout
+-----------------------
 
-        >>> from dso_api.dynamic_api.urls import router
-        >>> Model = router.all_models["dataset"]["tablename"]
-        >>> Model.objects.all()  # etc..
+Where a "viewset" binds all request handling, the "serializer" defines how the layout of the input/output
```
Comment (Yrob) [inline dev-docs/source/dynamic_api.rst:62]: Where a "viewset" binds all request handling, the "serializer" defines _what_ the layout of the input/output

File: dev-docs/source/dynamic_models.rst [hunk 2]
```diff
@@ -31,6 +31,28 @@ It builds a Django model based on the imported schema data.
...
+
+The schema files are imported by ``manage.py import_schemas``.
+This command reads all available schemas from a schema repository
+(:ref:`SCHEMA_URL <SCHEMA_URL>`, default: https://schemas.data.amsterdam.nl/datasets/)
+and updates the metadata tables accordingly.
+
+All schema data is saved in the :class:`~schematools.contrib.django.models.Dataset`
+model from the :mod:`schematools.contrib.django` package.
+Upon startup, DSO-API reads all available dataset schema's from
+the :class:`~schematools.contrib.django.models.Dataset` model to constructs the models.
```
Comment (Yrob) [inline dev-docs/source/dynamic_models.rst:47]: typo: `the :class:`~schematools.contrib.django.models.Dataset` model to _construct_ the models.`

File: dev-docs/source/streaming.rst [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+To avoid too many back/forth calls from the response-generator
+and the WSGI server, the produced output is submitted in chunks of 4096kB.
+Otherwrite an OS ``write()`` call might happen for a simple ``yield "}"`` statement.
+
+Next Link Optimization
+~~~~~~~~~~~~~~~~~~~~~~
+
+A standard paginator would do an expensive ``COUNT(*)`` on the table
+and use that to tell whether thare are additional pages.
```
Comment (Yrob) [inline dev-docs/source/streaming.rst:122]: and use that to tell whether _there_ are additional pages.

File: dev-docs/source/streaming.rst [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+We've optimized this by requesting one extra record from the database at the end of a page.
+This sentinel record is not rendered. It's existence does does indicate there is another page available.
```
Comment (Yrob) [inline dev-docs/source/streaming.rst:124]:
```suggestion
This sentinel record is not rendered. It's existence indicates there is another page available.
```

File: dev-docs/source/streaming.rst [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+This does mean the next/previous links have to be written at end of the response,
+after all main objects have been seen. This is handled by writing the main JSON object in chunks.
+
+Fields like a "total page count" and "total results" are no longer available, but rarely needed either.
+Typically clients only need a link to the next page.
+
```
Comment (Yrob) [inline dev-docs/source/streaming.rst:131]:
```suggestion
And we support the "_count" query parameter for when a client does need a result count. 
```

File: dev-docs/source/temporal.rst [hunk 6]
```diff
@@ -0,0 +1,159 @@
+Temporal Datasets
+=================
+
+Some datasets contain historical data. This allows you to navigate through previous versions of the same object.
+By default, the API only returns the current version of these objects.
+
+Navigating to other occurrences/versions is called "time travel in the
+`DSO standard <https://aandeslagmetdeomgevingswet.nl/digitaal-stelsel/aansluiten/standaarden/api-en-uri-strategie/>`_.
+Hence we speak of"temporal datasets".
```
Comment (Yrob) [inline dev-docs/source/temporal.rst:9]:
```suggestion
Hence we speak of "temporal datasets".
```

File: dev-docs/source/streaming.rst (hunk 4 above)
Comment (lbam) [reply dev-docs/source/streaming.rst:124]:
```suggestion
This sentinel record is not rendered. Its existence indicates that there is another page available.
```

File: dev-docs/source/streaming.rst (hunk 3 above)
Comment (lbam) [reply dev-docs/source/streaming.rst:122]: This is digging into implementation details far too deeply. Just say that determining the number of pages is too expensive. That's true for the remotes as well.

File: dev-docs/source/temporal.rst [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+      "properties": {
+        "identificatie": {"type": "string"},
+        "volgnummer": {"type": "integer"}
+      }
+    }
+
+If the "volgnummer" field is omitted, the relation only uses the first grouping field,
+and becomes a "loose relation".
+
+To make the relation versioned, add additional date fields can be added to the relation object:
```
Comment (Yrob) [inline dev-docs/source/temporal.rst:146]:
```suggestion
To make the relation versioned, additional date fields can be added to the relation object:
```

File: docs/source/generic/rest/fields.rst [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+    curl 'https://api.data.amsterdam.nl/v1/fietspaaltjes/fietspaaltjes/?_fields=-area,-noodzaak'
+
+.. note::
+    In plaats van ``_fields`` wordt ook ``fields`` ondersteund,
+    maar ``_fields`` heeft de voorkeur.
+
+.. note::
+    Het is niet mogelijk om velden tegelijk in te sluiten en uit te sluiten.
+    Het uitsluiten van velden is immers overbodig als je al de een lijst van specifieke velden opgeef.
```
Comment (Yrob) [inline docs/source/generic/rest/fields.rst:41]:
```suggestion
    Het uitsluiten van velden is immers overbodig als je al de een lijst van specifieke velden opgeeft.
```

File: docs/source/generic/rest/filtering.rst [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+Ieder veld kan gebruikt worden om op te filteren.
+Bijvoorbeeld:
+
+.. code-block:: bash
+
+    curl 'https://api.data.amsterdam.nl/v1/gebieden/stadsdelen/?naam=Westpoort'
+
+Als het veld een array van objecten is, kunnen de subvelden van de objecten
+filterd worden met de naam van de array en de naam van het subveld gescheiden
```
Comment (Yrob) [inline docs/source/generic/rest/filtering.rst:15]:
```suggestion
gefilterd worden met de naam van de array en de naam van het subveld gescheiden
```

File: docs/source/generic/rest/fields.rst (hunk 8 above)
Comment (lbam) [reply docs/source/generic/rest/fields.rst:41]:
```suggestion
```

No need to spell this out, better keep it short.

File: dev-docs/source/dynamic_api.rst [hunk 10]
```diff
@@ -1,11 +1,20 @@
 Dynamic API Endpoints
 =====================
 
+The project follows the standard Django REST Framework "router" logic
+together with viewsets.
+
+There is one twist.
 Since all :doc:`models are dynamically constructed <dynamic_models>`,
 the views, URL's and serializers are also dynamically constructed at startup.
```
Comment (lbam) [inline dev-docs/source/dynamic_api.rst:9]: Views and serializers, i.e., the instances, are always constructed at runtime. What we do differently is that the view and serializer *classes* are also dynamically constructed.

File: dev-docs/source/dynamic_api.rst [hunk 11]
```diff
@@ -1,11 +1,20 @@
...
-tools like Django's ``reverse()``, DRF's ``HyperlinkedRelatedField`` and the OpenAPI generator.
-Their code assumes there are hard-coded views and URLs in the project.
+
+Viewset Construction
+--------------------
+
+By dynamically constructing the viewsets (instead of using custom properties)
+the view logic integrates nicely with tools like Django's ``reverse()``,
+DRF's ``HyperlinkedRelatedField`` and the OpenAPI generator.
+Those classes all assume there are hard-coded views and URLs in the project.
```
Comment (lbam) [inline dev-docs/source/dynamic_api.rst:17]:
```suggestion
Those classes all assume there are hard-coded view classes and URLs in the project.
```

File: dev-docs/source/dynamic_models.rst [hunk 12]
```diff
@@ -31,6 +31,28 @@ It builds a Django model based on the imported schema data.
     :class:`~schematools.contrib.django.models.DatasetTable` and
     :class:`~schematools.contrib.django.models.DatasetField` models.
 
+Loading Schemas
+---------------
+
+To play with the examples, load the schema's first.
```
Comment (lbam) [inline dev-docs/source/dynamic_models.rst:37]:
```suggestion
To play with the examples, load the schemas first.
```

File: dev-docs/source/dynamic_api.rst (hunk 10 above)
Comment (vdboor) [reply dev-docs/source/dynamic_api.rst]: Good call! I've updated the intro even more. Should make it much easier to grasp the whole thing.

File: dev-docs/source/streaming.rst (hunk 5 above)
Comment (vdboor) [reply dev-docs/source/streaming.rst:133]: tx, added!

**Review by lbam — APPROVED** [review]

**Pull request merged**